### PR TITLE
Add Muse Glimmer 30B to the model zoo

### PR DESCRIPTION
**Pull request by CharlesCNorton**
## 🔗 Related Issues

No related issues to link

## 📋 What changes are proposed in this pull request?

Adds `muse-glimmer-30b-torch` ([meta-models/Muse-Glimmer-30B](https://huggingface.co/meta-models/Muse-Glimmer-30B), Apache 2.0) to the torch model zoo for zero-shot object detection via 2D grounding, with a `fiftyone/utils/muse_glimmer.py` wrapper following the Qwen3-VL pattern.

- The model emits reasoning on a `to=self` channel before answering on a `to=user` channel. The output processor parses only the answer channel; a generation that truncates before it yields an empty result.
- Detections are `{"label", "bbox_2d"}` JSON on a 0-1000 scale, converted to relative `[x, y, w, h]`.
- The language model loads with 4-bit NF4 quantization by default, fitting 24 GB cards; the vision tower remains bf16 because its forward casts pixels to its own weight dtype. `load_in_4bit=False` loads full bf16 (59.6 GB).

## 🧪 How is this patch tested? If it is not, please explain why.

- `tests/unittests/utils/test_muse_glimmer.py`: 22 tests covering answer-channel extraction, truncation handling, bbox parsing and clamping, and config defaults. No GPU or model download required.
- Live on an RTX 6000 Ada: NF4 load in 51 s at 18.7 GB VRAM; a COCO validation image produced 30 detections with correct geometry under greedy decoding. The captured transcript confirms the channel structure and box format the parser targets.

### Is this a user-facing change that should be mentioned in the release notes?

- [x] Yes. Give a description of this change to be included in the release notes for FiftyOne users.

Added `muse-glimmer-30b-torch` (Muse Glimmer 30B) to the model zoo for zero-shot object detection via 2D grounding.

### What areas of FiftyOne does this PR affect?

- [x] Core: Core `fiftyone` Python library changes

## Summary by CodeRabbit

- **New Features**
  - Added Muse Glimmer model support to the FiftyOne Model Zoo.
  - Supports configurable prompts, class lists, image generation, channel-aware decoding, and optional 4-bit quantization.
  - Converts valid model responses into FiftyOne detections while filtering invalid or incomplete results.

- **Tests**
  - Added coverage for response parsing, bounding-box validation, batch processing, and custom model configuration.

---
**Enterprise sync PR**: https://github.com/voxel51/fiftyone-teams/pull/3699

### CODE REVIEW

**Review by benjaminpkane — APPROVED** [review]

File: fiftyone/utils/muse_glimmer.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+import torch
+
+logger = logging.getLogger(__name__)
+
+
+def _ensure_muse_glimmer():
+    fou.ensure_package("transformers>=5.15.0")
+    fou.ensure_package("accelerate")
+    # 4-bit NF4 loading; the model is 59.6 GB at bf16
+    fou.ensure_package("bitsandbytes")
```
[the remark this answers is not in the document]
Comment (CharlesCNorton) [reply fiftyone/utils/muse_glimmer.py]: Fixed in ee6f015160. `bitsandbytes` is no longer ensured by the lazy `transformers` import or listed in the manifest's packages; `_load_model` ensures it only for 4-bit loading on GPU. `MuseGlimmerOutputProcessor.__call__` now takes `classes` and keeps only detections whose label is in it, the same exact-match filter the core output processors apply, so `filter_classes` takes effect. `test_classes_filter_detections` covers the filter end to end and `test_bitsandbytes_is_not_required_up_front` covers the import requirements.
